build.gradle.kts

```diff
@@ -211,25 +229,30 @@ subprojects {
   tasks.withType<Test>().configureEach {
     environment("OKHTTP_ROOT", rootDir)
   }
+  tasks.withType<KotlinJvmTest>().configureEach {
+    environment("OKHTTP_ROOT", rootDir)
+  }
 
-  if (platform == "jdk8alpn") {
-    // Add alpn-boot on Java 8 so we can use HTTP/2 without a stable API.
-    val alpnBootVersion = alpnBootVersion()
-    if (alpnBootVersion != null) {
-      val alpnBootJar = configurations.detachedConfiguration(
-        dependencies.create("org.mortbay.jetty.alpn:alpn-boot:$alpnBootVersion")
-      ).singleFile
-      tasks.withType<Test> {
-        jvmArgs("-Xbootclasspath/p:${alpnBootJar}")
+  if (project.name != "okhttp") {
+    if (platform == "jdk8alpn") {
+      // Add alpn-boot on Java 8 so we can use HTTP/2 without a stable API.
```

Comment:
Does this make sense for non-okhttp modules?

Reply:
Good call.

I do think our grade files could do with a major redesign/simplification. Maybe less build variants overall.

Each thing like osgi, multiplatform, loom, graal has made them more special case. And extra test runs and platforms to test.

Reply:
https://github.com/square/okhttp/pull/8785